drivers/counter/CMakeLists.txt

```
@@ -48,3 +48,4 @@ zephyr_library_sources_ifdef(CONFIG_COUNTER_TIMER_GD32		counter_gd32_timer.c)
 zephyr_library_sources_ifdef(CONFIG_COUNTER_SNPS_DW             counter_dw_timer.c)
 zephyr_library_sources_ifdef(CONFIG_COUNTER_SHELL               counter_timer_shell.c)
 zephyr_library_sources_ifdef(CONFIG_COUNTER_TIMER_RPI_PICO	counter_rpi_pico_timer.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER_MSPM0GXXXX	        counter_mspm0gxxxx.c)
```

TI started renaming these (finally ^^). So maybe better to make this patch compatible right away :)
```suggestion
zephyr_library_sources_ifdef(CONFIG_COUNTER_MSPM0	        counter_mspm0.c)
```